extensions/smallrye-jwt/deployment/src/test/java/io/quarkus/jwt/test/dev/SmallryeJwtProcessorDevServiceProcessorTest.java

```diff
@@ -0,0 +1,77 @@
+package io.quarkus.jwt.test.dev;
+
+import jakarta.annotation.security.PermitAll;
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.Produces;
+import jakarta.ws.rs.core.MediaType;
+
+import org.eclipse.microprofile.jwt.Claims;
+import org.hamcrest.Matchers;
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.jwt.test.GreetingResource;
+import io.quarkus.test.QuarkusDevModeTest;
+import io.restassured.RestAssured;
+import io.restassured.http.Header;
+import io.smallrye.jwt.build.Jwt;
+
+public class SmallryeJwtProcessorDevServiceProcessorTest {
+
+    @RegisterExtension
+    static QuarkusDevModeTest devMode = new QuarkusDevModeTest()
+            .withApplicationRoot((jar) -> jar.addClasses(
+                    GreetingResource.class, TokenResource.class).addAsResource(
+                            new StringAsset(""), "application.properties"));
+
+    @Test
+    public void shouldNotBeNecessaryToAddSignKeysOnApplicationProperties() {
+        String token = RestAssured.given()
+                .header(new Header("Accept", "text/plain"))
+                .get("/token")
+                .andReturn()
+                .body()
+                .asString();
+
+        RestAssured.given()
+                .header(new Header("Authorization", "Bearer " + token))
+                .get("/only-user")
+                .then().assertThat().statusCode(200);
+    }
+
+    @Test
+    public void shouldUseTheSameKeyPairOnLiveReload() {
+        String token = RestAssured.given()
+                .header(new Header("Accept", "text/plain"))
+                .get("/token")
+                .andReturn()
+                .body()
+                .asString();
+
+        devMode.modifySourceFile("GreetingResource.java", s -> s.replace("Hello from Quarkus", "Hello from JWT"));
+
+        // there is no need to get another token
+        RestAssured.given()
+                .header(new Header("Authorization", "Bearer " + token))
+                .get("/only-user")
+                .then().assertThat().statusCode(200)
+                .body(Matchers.containsString("Hello from JWT"));
+    }
+
+    @Path("/token")
```

Comment:
Why would you like to keep it as a remote call, the token should be generated without problems directly in the test code

Reply:
I think it shouldn't matter whether it is inside of the resource or not. Are you sure it wasn't just timing? Can you replace the endpoint call with that `Jwt...` and confirm it still happens?

Reply:
@michalvavrik @mcruzdev As far as I recall, properties registered in devservice processors are not available at the test level...
Hmm, I 100% certain I typed a suggestion here yesterday from my phone, looks like I lost it...

Here is what I tried to type yesterday...
Have a look at `KeycloakTestClient`, it implements `DevServicesContext.ContextAware`, this is where `KeycloakTestClient` picks up properties from even in native mode... Tests access `KeycloakTestClient` as a non static test field.

IMHO having a class such as `class JwtTokenKeys implements DevServicesContext.ContextAware` and then tests doing `Jwt....sign(jwtTokenkeys.getSignPrivateKey())` would be better, instead of asking users to create a REST endpoint to get the token generated - which would offset the benefit of avoiding dealing with generating keys...

Reply:
@mcruzdev For now, you can only update the existing devmode test, and then, once everyone is happy with the rest of the PR, I can suggest updates to one of the integration tests which don't use devmode

Reply:
> For now, you can update the existing devmode test only if you'd like

right @mcruzdev @sberyozkin , that is what I was missing! this is DEV mode test, so of course you cannot access config properties set inside DEV mode. I thought that comments above doesn't add-up. You see, config properties set via Dev SVC are config properties like any other. So if this was `QuarkusUnitTest` it would work.

On that note, you should rename that test to make it clear it is DEV mode test. Thanks

Reply:
that doesn't contradict Sergey's suggestion, I am just saying this is expected behavior.